ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java

```diff
@@ -282,7 +283,9 @@ public GraphQLDataFetchers(final Set<Capability> supportedCapabilities) {
   private TransactionAdapter getTransactionAdapter(
       final TransactionWithMetadata transactionWithMetadata) {
     final Transaction transaction = transactionWithMetadata.getTransaction();
-    return goQuorumPrivacyParameters.isPresent() && transaction.isGoQuorumPrivateTransaction()
+    final boolean isGoQuorumPrivateTransaction =
+        transaction.isGoQuorumPrivateTransaction(GoQuorumOptions.goQuorumCompatibilityMode);
```

Comment:
Can the check of `GoQuorumOptions.goQuorumCompatibilityMode` be skipped and we presume that if the `goQuorumPrivacyParameters` are set that we need to goQuorum decode it?